peer.go

```diff
@@ -441,6 +440,7 @@ func newChanMsgStream(f *fundingManager, h *htlcswitch.Switch, p *peer,
 
 // Start starts the chanMsgStream.
 func (c *chanMsgStream) Start() {
+	c.wg.Add(1)
```

Comment:
Hopefully! Having a pattern is the important part 🙂

Forgot to mention! The approach above also allows you to mindlessly avoid accidentally capturing variables in scope that mutate, e.g. when iterating over a slice, since it forces you to pass in the parameters as arguments.

In some cases it may be useful to capture local variables and not have a tiny function with 8 arguments. There's never a one size fits all solution, but just some things to keep in mind if you have options :)